psydac/feec/derivatives.py

```diff
@@ -1,121 +1,248 @@
 # -*- coding: UTF-8 -*-
 
 import numpy as np
-from numpy import zeros
-from numpy import concatenate
-from numpy import block
 
-from scipy import kron
-from scipy.sparse import identity
-from scipy.sparse import csr_matrix
+from psydac.linalg.stencil import StencilMatrix, StencilVectorSpace
+from psydac.linalg.kron    import KroneckerStencilMatrix as Stencil_kron
+from psydac.linalg.block   import ProductSpace, BlockVector, BlockLinearOperator, BlockMatrix
+from psydac.fem.vector     import ProductFemSpace
 
-def d_matrix(n):
+def d_matrix(n, p, P):
```

Comment:
Why is this function called `d_matrix`? Does the `d` in the name have to do with the *exterior derivative*?

And how is this function used in practice? Which object is going to call it?

More explanation in the docstring would be very helpful.

Reply:
the letter d here means derivative